word-search.py

```diff
@@ -229,8 +229,39 @@ def checkWord():
     # checks if user found a correct word
     return
 
-def printWordList():
-    # prints all possible words to be found at game end
+def printWordList(gridWordList, foundWords):
+    # Prints all possible words to be found at game end
+    # Reveals player's answers and possible answers
+    
+    from collections import defaultdict
+
+    # Group words by their lengths from gridWordList
+    length_groups = defaultdict(list)
+    for word in gridWordList:
+        if len(word) >= 3:  # Only consider words with length 3 or more
+            length_groups[len(word)].append(word)
+
+    # Sort the words within each group
+    for length in length_groups:
+        length_groups[length].sort()
+
+    # Separate the words into possible and found words
+    possible_words = defaultdict(list)
+    for length in length_groups:
+        for word in length_groups[length]:
+            if word not in foundWords:
+                possible_words[length].append(word)
+
+    # Print possible words
+    print("\nHere are the possible words:")
+    for length in possible_words:
+        print(" ".join(possible_words[length]))
```

Comment:
This is good, but can we also include a header/label to indicate which list we are printing? For example:
<img width="1249" alt="Screenshot 2024-12-10 at 3 34 23 PM" src="https://github.com/user-attachments/assets/4012565d-90bb-41c3-b799-ea93ac2a3e65">

Just to make the distinction clearer to the user :)